ADALiOS/ADALiOS/ADProfileInfo.m

```diff
@@ -0,0 +1,269 @@
+// Copyright © Microsoft Open Technologies, Inc.
+//
+// All Rights Reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+// ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+// PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache License, Version 2.0 for the specific language
+// governing permissions and limitations under the License.
+
+#import "ADProfileInfo.h"
+#import "ADALiOS.h"
+#import "ADOAuth2Constants.h"
+#import "NSString+ADHelperMethods.h"
+
+static NSString* const PROFILE_INFO_SUBJECT = @"sub";
+static NSString* const PROFILE_INFO_OBJID = @"oid";
```

Comment:
Rejected! We need old right now because sub isn't implemented at the server. I agree we should remove `oid` when we can and I'll create a tracking bug for this.